analysis_driver/tool_versioning.py

```diff
@@ -122,6 +121,7 @@ def _get_stdout(cmd):
         return stdout.strip().decode('utf-8')
 
     def __getitem__(self, item):
+        self.add_tool(item)
```

Comment:
The issue with this implementation is that it also lazily check the version. 
If we have the wrong version it might crash 2 days into the run.
Not a major problem since we control the versions in use but still.

Reply:
Also if we resume a sample, the tool versions file will be overwritten, potentially with missing tools from skipped stages. We could keep a register of all tools that a pipeline should use, but we'd then need to keep that up to date. Or we can remove the option of multiple pipelines sharing toolsets, and make it a 1 pipeline - 1 toolset relationship

Reply:
Update: toolset now loads, updates and rewrites the file if it already exists.